apps/sms/views/conversation/js/conversation.js

```diff
@@ -1955,67 +1843,39 @@ var ConversationView = {
-    // - move to thread list if needed
-
-    if (!Array.isArray(list)) {
-      list = [list];
-    }
-
-    // Removing from DOM all messages to delete
-    for (var i = 0, l = list.length; i < l; i++) {
-      Threads.unregisterMessage(list[i]);
-      ConversationView.removeMessageDOM(
-        document.getElementById('message-' + list[i])
-      );
-    }
+  /**
+   * Removes selected messages from DB and view (DOM), updates conversation in
+   * inbox without re-rendering and moves to the inbox in case all messages were
+   * removed.
+   * @param {Array.<Number>} messageIdList List of message ids to remove.
+   * @returns {Promise}
+   */
+  deleteMessages: function conv_deleteMessages(messageIdList) {
+    return MessageManager.deleteMessages(messageIdList).then(() => {
+      // Removing from DOM all messages to delete
+      messageIdList.forEach((messageId) => {
+        Threads.unregisterMessage(messageId);
+        this.removeMessageDOM(document.getElementById('message-' + messageId));
+      });
 
-    callback = typeof callback === 'function' ? callback : function() {};
+      // Did we remove all messages of the conversation?
+      if (!this.activeThread.messages.size) {
+        return this.backOrClose();
+      }
 
-    // Do we remove all messages of the Thread?
-    if (!ConversationView.container.firstElementChild) {
-      callback();
-      this.backOrClose();
-    } else {
-      // Retrieve latest message in the UI
-      var lastDateGroup = ConversationView.container.lastElementChild;
+      // Retrieve latest message in the UI.
+      var lastDateGroup = this.container.lastElementChild;
       var lastMessage = lastDateGroup.querySelector('li:last-child');
-      // We need to make thread-list to show the same info
-      var request = MessageManager.getMessage(+lastMessage.dataset.messageId);
-      request.onsuccess = function() {
-        callback();
-        InboxView.updateThread(request.result, { deleted: true });
-      };
-      request.onerror = function() {
-        console.error('Error when updating the list of threads');
-        callback();
-      };
-    }
-  },
 
-  delete: function conv_delete() {
-    function performDeletion() {
-      /* jshint validthis: true */
-
-      WaitingScreen.show();
-      var items = this.selectionHandler.selectedList;
-      var delNumList = items.map(item => +item);
-
-      // Complete deletion in DB and in UI
-      MessageManager.deleteMessages(delNumList,
-        function onDeletionDone() {
-          ConversationView.deleteUIMessages(delNumList, function() {
-            ConversationView.cancelEdit();
-            WaitingScreen.hide();
-          });
-        }
+      // We need to make inbox to show the same info.
+      return MessageManager.getMessage(+lastMessage.dataset.messageId).then(
+        (message) => InboxView.updateThread(message, { deleted: true }),
+        (e) => console.error('Error when updating the list of threads', e)
       );
```

Comment:
just wondering if we should return the result of getMessage and move this `then` to the outer promise chain ?

Also this would catch any error coming from MessageManager.deleteMessages